xmppserver/src/main/java/org/jivesoftware/util/cache/ConsistencyChecks.java

```diff
@@ -720,8 +720,7 @@
             .collect(Collectors.toList());
         final List<String> allOccupantsJids = allOccupants
             .stream()
-            .map(OccupantManager.Occupant::getRealJID)
-            .map(JID::toFullJID)
+            .map(occupant -> occupant.getRealJID().toFullJID() + "/" + occupant.getNickname())
```

Comment:
This is weird to me. It combines a 'real jid' of an occupant, which is suggested to be a full JID, and adds a nickname to that.

So, in an example, giving a user's real JID `john@example.org/mobilephone` that is in a chatroom `myroom@conference.example.org` using nickname `Johnny` (this would make their "room JID" `myroom@conference.example.org/Johnny`) this value would be `john@example.org/mobilephone/Johnny?

If this is the intended end result (and to be used only to distinguish between entries), then this is OK. I would still suggest to use another way to concatenate `real JID` and `nickname` together, to make that more obvious. Something like `john@example.org/mobilephone with nickname: Johnny` for example.

My concern is that the Full JID in reality turns out to be a bare JID, which is then "completed" with the nickname. That would result in a full JID that mixes parts of the real JID and the room JID. That would be a JID that is not usable to route messages.

Reply:
Valid concern. Intention was not to create a new JID, but to clarify why the same occupant JID could be in the list of occupants multiple times. I now improved the code to not suggest constructing some new JID.